packages/host/app/components/matrix/room.gts

```diff
@@ -32,6 +36,33 @@ interface RoomArgs {
     roomId: string;
   };
 }
+
+interface CommandArgs {
+  Args: {
+    command: any;
+    onCommand: (command: any) => void;
+  };
+}
+
+class CommandMessage extends Component<CommandArgs> {
+  @service private declare commandService: CommandService;
+  <template>
+    <Button data-test-command-apply {{on 'click' this.clicked}}>Apply</Button>
+  </template>
+
+  @action
+  async clicked() {
+    await this.commandService.runCommand(
+      this.args.command.type,
+      this.args.command,
+    );
+  }
+
+  constructor(owner: unknown, args: any) {
+    super(owner, args);
+  }
```

Comment:
probs can remove this if not doing any logic in constructor

Reply:
confirm please delete this constructor